pyi.py

```diff
@@ -270,6 +270,27 @@ def _is_bad_TypedDict(node: ast.Call) -> bool:
     )
 
 
+@dataclass
+class AttributeManager:
```

Comment:
I don't like the name, but I can't think of a good name. How about `NestingCounter`?